File: backend/Domain/Dtos/Responses/LoginResponse.cs
using System.Text.Json.Serialization;
using Domain.Entities.Base;

namespace Domain.Dtos.Responses;

public class Login<TResponse> : Response
{
	[JsonPropertyName("Response")]
	public TResponse _Response { get; set; }
	public AuthToken AuthToken { get; set; }
	public RefreshToken RefreshToken { get; set; }

	public static Login<TResponse> Success(TResponse response, AuthToken authToken, RefreshToken refreshToken) =>
		new Login<TResponse>
		{
			IsSuccessful = true,
			_Response = response,
			AuthToken = authToken,
			RefreshToken = refreshToken
		};
}

File: frontend/src/app/models/responses/login-response.ts
import { AuthToken } from "../auth/auth-token";
import { RefreshToken } from "../auth/refresh-token";
import { Response } from "../base/response";

export class LoginResponse<TEntity> extends Response {
	response: TEntity;
	authToken: AuthToken;
	refrehToken: RefreshToken;
}
